### PR TITLE
New chem, saline-glucose. Adjusts isotonic mix.

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
image
Changes isotonic from restoring nutrition, to purely restoring blood volume. Don't forget your MRE!
Removes iron regenerating blood.
Adds new chem saline glucose, restores 1.2 units of blood.
image
Recipe is salt, water and sugar.
image

## Why It's Good For The Game
Blood restore pills restore blood, not food.
This makes it so isotonics no longer invalidate MRE in every way.
## Changelog
:cl:
add: New chem, saline glucose, restores blood at a rate of 1.2 units (QC is 0.2, Nano is 2.4)
del: Iron no longer regens blood.
add: Saline-Glucose has replaced iron in the cryomix.
refactor: Isotonics are now just 15u of saline-glucose. Pills hold 1 dose, injectors hold 2 doses.
/:cl:

### CODE REVIEW

**Comment by Andrewhedge** [comment]
wont this ruin QC, my good mate? also as elsa pointed out, blood-filling IV nontheless le blood type. this is something

File: code/modules/reagents/reagents/medical.dm [hunk 1]
```diff
@@ -275,6 +275,26 @@
...
 	L.apply_damages(3*effect_str, 0, 3*effect_str)
 
+/datum/reagent/medicine/saline_glucose
+	name = "Saline-Glucose"
+	description = "Saline-Glucose can be used to restore blood in a pinch."
+	color = "#d4f1f9"
+	overdose_threshold = REAGENTS_OVERDOSE
+	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL
+	taste_description = "salty water"
+
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:285]: Make it scannable.

Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:285]: Make it scannable.

File: code/modules/reagents/reactions/medical.dm [hunk 2]
```diff
@@ -1,3 +1,9 @@
+/datum/chemical_reaction/saline_glucose
+	name = "Saline-Glucose" //Chem to restore blood.
+	results = list(/datum/reagent/medicine/saline_glucose = 3)
+	required_reagents = list(/datum/reagent/consumable/sodiumchloride = 1, /datum/reagent/water = 1, /datum/reagent/consumable/sugar = 1)
+	mob_react = FALSE
```
Comment (QualityVan) [inline code/modules/reagents/reactions/medical.dm:5]: Why is denying mob_react important here?

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 3]
```diff
@@ -276,14 +276,13 @@
...
-	amount_per_transfer_from_this = 25
-	volume = 25
+	amount_per_transfer_from_this = 15
+	volume = 30
 	list_reagents = list(
-		/datum/reagent/iron = 10,
-		/datum/reagent/consumable/nutriment = 10,
-		/datum/reagent/consumable/sugar = 5,
+		/datum/reagent/iron = 15,
+		/datum/reagent/medicine/saline_glucose = 15,
```
Comment (QualityVan) [inline code/game/objects/items/reagent_containers/autoinjectors.dm:285]: Do you think marine blood regen rate is too low at the moment? You're increasing it by a third with the 'default' mix.

Comment (Xander3359) [reply code/game/objects/items/reagent_containers/autoinjectors.dm:285]: I think isotonics should be purely about regenning blood and not just nutrition, and yes, I think even chugging the fuck out of isotonics still regens blood very slow.

**Comment by Xander3359** [comment]
> How is it different to iron though, gimme the numbers please lad

It's not different to iron, it's meant to be used alongside it.
image
image

**Comment by QualityVan** [comment]
My only remaining concern is that this bumps an injector from one shot of fifty blood (10 iron, 4 blood per iron, 10 nutriment at 1:1) to two shots of 90 (15u salglu per shot, 6 blood per unit). That's almost quadruple, and without the old limiter of needing to watch food level (which, admittedly, was the point of this pr in the first place). It might be worth bumping the metab rate to 0.4 which would make it only double the old efficiency. I am, however, being picky here.

**Comment by Xander3359** [comment]
Well I do want it to heal more blood as I feel like the old isotonic mix is sad when it comes to regenning blood, but I've put in the custom metabolism as per your suggestion.

**Comment by Lumipharon** [comment]
I'm defering to QV's opinions about this btw. Nanoblood has pretty severe side effects and is quite rare, while this is infinite, free and basically has no side effects (and of course you can stack it with nano), so it really shouldn't be particularly strong considering you can have 240u of it in the same space as your nanoblood hypo, and you can just chug it all day long.

**Review by QualityVan — APPROVED** [review]
Fine as is. More punitive crit OD is optional but not critical (default OD is fine, doesn't need to be any worse), same with letting regen continue past base levels.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**